packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -231,16 +231,12 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
       if (current != null) {
         if (current is TrainHoppingAnimation) {
           TrainHoppingAnimation newAnimation;
-          newAnimation = new TrainHoppingAnimation(
-            current.currentTrain,
-            nextRoute._animation,
-            onSwitchedTrain: () {
-              assert(_secondaryAnimation.parent == newAnimation);
-              assert(newAnimation.currentTrain == nextRoute._animation);
-              _secondaryAnimation.parent = newAnimation.currentTrain;
-              newAnimation.dispose();
-            }
-          );
+          newAnimation = new TrainHoppingAnimation(current.currentTrain, nextRoute._animation, onSwitchedTrain: () {
```

Comment:
nit: it would probably be better to revert the untouched lines to the original formatting.